packages/navigator/src/browser/open-editors-widget/navigator-open-editors-tree-model.ts

```diff
@@ -0,0 +1,218 @@
+
+    protected toDisposeOnPreviewWidgetReplaced = new DisposableCollection();
+
+    protected _editorWidgetsByArea = new Map<ApplicationShell.Area, NavigatableWidget[]>();
+
+    get editorWidgets(): NavigatableWidget[] {
+        let editorWidgets: NavigatableWidget[] = [];
+        this._editorWidgetsByArea.forEach(widgets => {
+            editorWidgets = [...editorWidgets, ...widgets];
+        });
+        return editorWidgets;
+    }
+
+    getEditorWidgetsByGroup(id: ApplicationShell.Area | number): NavigatableWidget[] | undefined {
+        const idAsNum = Number(id);
+        if (!isNaN(idAsNum)) {
+            return this._editorWidgetsByGroup.get(idAsNum);
+        }
+        return this._editorWidgetsByArea.get(id as ApplicationShell.Area);
+    }
+
+    protected _editorWidgetsByGroup = new Map<number, NavigatableWidget[]>();
+
+    @postConstruct()
+    protected init(): void {
+        super.init();
+        this.setupHandlers();
+        this.initializeRoot();
+    }
+
+    protected setupHandlers(): void {
+        this.toDispose.push(this.selectionService.onSelectionChanged(selection => {
+            const { widget } = (selection[0] as OpenEditorNode);
+            this.applicationShell.activateWidget(widget.id);
+        }));
+        this.toDispose.push(this.applicationShell.onDidChangeCurrentWidget(async ({ newValue }) => {
+            const nodeToSelect = this.tree.getNode(newValue?.id) as SelectableTreeNode;
+            if (nodeToSelect) {
+                this.selectNode(nodeToSelect);
+            }
+        }));
+        this.toDispose.push(this.workspaceService.onWorkspaceChanged(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.workspaceService.onWorkspaceLocationChanged(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.applicationShell.onDidAddWidget(widget => {
+            if (ApplicationShell.TrackableWidgetProvider.is(widget) && widget.onDidChangeTrackableWidgets) {
+                this.toDisposeOnPreviewWidgetReplaced.dispose();
+                this.toDisposeOnPreviewWidgetReplaced.push(widget.onDidChangeTrackableWidgets(() => this.updateOpenWidgets()));
+            }
+            this.updateOpenWidgets();
+        }));
+        this.toDispose.push(this.applicationShell.onDidRemoveWidget(() => this.updateOpenWidgets()));
+        this.applicationShell.mainPanel.layoutModified.connect(() => this.updateOpenWidgets());
+        this.applicationShell.bottomPanel.layoutModified.connect(() => this.updateOpenWidgets());
+    }
+
+    protected async initializeRoot(): Promise<void> {
+        await this.updateOpenWidgets();
+        this.fireChanged();
+    }
+
+    protected updateOpenWidgets = debounce(this.doUpdateOpenWidgets, 250);
+
+    protected async doUpdateOpenWidgets(): Promise<void> {
+        this._editorWidgetsByArea = new Map<ApplicationShell.Area, NavigatableWidget[]>();
+        const areas: ApplicationShell.Area[] = ['main', 'bottom', 'left', 'right'];
```

Comment:
I'm sure you were tempted to change this to an `enum` rather than a string union type. Maybe it should be done? I also notice you've omitted `top`. Is that intentional?

Reply:
I had thought about that, but there are enough placed where this type is used to make me a bit nervous to do that now. I did intentionally leave 'top' out because I believe that area refers to the browser versions top tabbar (File menu) and didn't expect to find any widgets there. But I've added it back in since it shouldn't really make any difference.